apps/explorer/lib/explorer/smart_contract/reader.ex

```diff
@@ -356,39 +354,87 @@ defmodule Explorer.SmartContract.Reader do
     "tuple[#{tuple_types}]"
   end
 
+  @spec fetch_current_value_from_blockchain(
+          any(),
+          [%{optional(binary()) => any()}],
+          Explorer.Chain.Hash.t(),
+          boolean(),
+          keyword(),
+          nil | binary()
+        ) :: [SmartContract.function_description()]
   def fetch_current_value_from_blockchain(
-        function,
+        functions,
         abi,
         contract_address_hash,
         leave_error_as_map,
         options,
         from \\ nil
       ) do
-    case function do
-      %{"inputs" => []} ->
-        method_id = function["method_id"]
-        args = function["inputs"]
-
-        %{output: outputs, names: names} =
-          query_function_with_names(
-            contract_address_hash,
-            %{method_id: method_id, args: args},
-            :regular,
-            from,
-            abi,
-            leave_error_as_map,
-            options
-          )
-
-        function
-        |> Map.replace!("outputs", outputs)
-        |> Map.put("abi_outputs", Map.get(function, "outputs", []))
-        |> Map.put("names", names)
+    initial_methods_id_order = Enum.map(functions, &Map.get(&1, "method_id"))
+
+    %{to_be_fetched: to_be_fetched, method_id_to_outputs: method_id_to_outputs, unchanged: unchanged} =
+      Enum.reduce(functions, %{to_be_fetched: %{}, method_id_to_outputs: %{}, unchanged: %{}}, fn function,
+                                                                                                  %{
+                                                                                                    to_be_fetched:
+                                                                                                      to_be_fetched,
+                                                                                                    unchanged:
+                                                                                                      unchanged,
+                                                                                                    method_id_to_outputs:
+                                                                                                      method_id_to_outputs
+                                                                                                  } ->
```

Comment:
It's a matter of taste, but what about changing formatting here to avoid long lines?

I would suggest something like this:

```suggestion
  Enum.reduce(
    functions,
    %{to_be_fetched: %{}, method_id_to_outputs: %{}, unchanged: %{}},
    fn function,
       %{
         to_be_fetched: to_be_fetched,
         unchanged: unchanged,
         method_id_to_outputs: method_id_to_outputs
       } ->
```

Reply:
Good suggestion